modal/cls.py

```diff
@@ -292,7 +292,19 @@ def with_options(
         if gpu or cpu or memory:
             milli_cpu = int(1000 * cpu) if cpu is not None else None
             gpu_config = parse_gpu_config(gpu)
-            resources = api_pb2.Resources(milli_cpu=milli_cpu, gpu_config=gpu_config, memory_mb=memory)
+            if memory and isinstance(memory, int):
+                memory_mb = memory
+                memory_mb_max = 0  # no limit
+            elif memory and isinstance(memory, tuple):
+                memory_mb, memory_mb_max = memory
+                if memory_mb_max < memory_mb:
+                    raise ValueError(f"Cannot specify a memory limit lower than request: {memory_mb_max} < {memory_mb}")
```

Comment:
It's a little unclear to me whether we have a policy of always raising `modal.exception` classes from the client or whether that is even a good idea, but I think that usually we would raise an `InvalidError` here? Although `ValueError` is also correct.

Reply:
Yeh also thought the same. I think `ValueError` is correct, but I'd also want to things to be consistently correct across the codebase.

Reply:
I'm sure this isn't the only instance; maybe we should just make a ticket and try to standardize prior to v1.0